### DISCUSSION TITLE
New IdeaVim extensions released: FunctionTextObj, Switch, Peekaboo

### DISCUSSION THREAD

**Discussion by jphalip** [JetBrains/ideavim#1047] [General]
Hi there,

For those interested, I just wanted to mention a few extensions I've just released on the marketplace:

* [FunctionTextObj](https://plugins.jetbrains.com/plugin/25897-vim-functiontextobj): Adds the ability to refer to functions/methods as text objects, so you can do things like `cif`, `vaf`, `yaf`.
* [Switch](https://plugins.jetbrains.com/plugin/25899-vim-switch): Allows to quickly switch a term or expression with related ones (e.g. `true`/`false`, `protected`/`private`/`public`, etc.). It comes with some built-in definitions and you can also create your own.
* [Peekaboo](https://plugins.jetbrains.com/plugin/25776-vim-peekaboo): Shows the contents of your registers in a popup window when you press `"` in normal mode and `ctrl-r` in insert mode.

Hope you find it useful. Feedback welcome!

Julien

**Comment by AlexPl292** [comment]
Cool! If you want, you can add them [here](https://github.com/JetBrains/ideavim/blob/master/doc/IdeaVim%20Plugins.md).

**Comment by jphalip** [comment]
@AlexPl292 Sure! Here's the PR: https://github.com/JetBrains/ideavim/pull/1051

**Comment by huyz** [comment]
Oh great. I've been needing some of that
